tests/test_arrow_dataset.py

```diff
@@ -0,0 +1,114 @@
+import os
+import tempfile
+from unittest import TestCase
+
+import numpy as np
+import pyarrow as pa
+
+from nlp.arrow_reader import BaseReader
+from nlp.info import DatasetInfo
+from nlp.splits import SplitDict, SplitInfo
+
+
+class ReaderTester(BaseReader):
```

Comment:
@lhoestq - it's slightly different from what you used so I copy pasted the code. I guess we could refactor this into `test_utils.py` at one point.